app/src/main/java/org/mifos/mobilebanking/ui/fragments/RegistrationFragment.java

```diff
@@ -115,61 +151,153 @@ public void registerClicked() {
     }
 
     private boolean areFieldsValidated() {
```

Comment:
@letelete What you are doing is correct. But instead of creating methods for every `if`, write them in a single method like it was written before.

Reply:
@miPlodder **Isn't it a bad practice to create long methods?** 

It's so unreadable and hard to maintain if the single method contains all of the validation logic. It's also easy to repeat yourself here, and that's what I've seen in your code before my fix. There were 2 validations for some of the input (I guess it was a password) which was checking if password length is greater than a) 5, b) 6 which is complete nonsense and it 100% comes out of hard-to-read method with tons of validations. I've also seen validations for some specific inputs which were not grouped-up so at the beginning I could spot X validation and at the end of the method, I also spot validation for the same X input.

**My solution ensures clean-code standards.** Methods are easy to maintain now and if we want to change some of the validation we just need to search for a specific method instead of analyzing one, enormous method.

**Splitting validations into method has also another positive aspect**, which is a fact, that with the live-update feature, we don't need to check every single validation and just the one we are interested in(validation for currently edited input).

Reply:
@letelete I know that your code is cleaner. But I have a better alternative which I will add to the GCI tasks in future. 
We can create a Utils Class where we can add all the validations methods (like you added here). By that way, we can call those methods, for any of the EditText (like Sign up, Login) where that validation is required. This will reduce redundant code for validation.

Reply:
Yeah, that's actually a good idea which I was thinking about while faced with writing the same code for login_activity validation(#911) 😆 I will get all of these and split them into one as it was before, then.